app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,23 @@
+class BookingsController < ApplicationController
+  before_action :set_booking, only: %i[cancel]
+
+  def create
+    @event = Event.find(params[:event_id])
+    @booking = @event.bookings.new
+    @booking.user = current_user
+
+    redirect_to root_path if @booking.save
+  end
+
+  def cancel
+    @booking = Booking.find(params[:id])
+    @booking.update(canceled: true)
+    redirect_back fallback_location: root_path
+  end
```

Comment:
Why are you finding the booking inside cancel if you already have the set_booking method ?